pkg/prober/prober.go

```diff
@@ -0,0 +1,158 @@
+package prober
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/improbable-eng/thanos/pkg/component"
+	"github.com/prometheus/common/route"
+)
+
+const (
+	healthyEndpointPath  = "/-/healthy"
+	readyEndpointPath    = "/-/ready"
+	okProbeText          = "thanos %v is %v"
+	errorProbeText       = "thanos %v is not %v. Reason: %v"
+	probeErrorHTTPStatus = 503
+	initialErrorText     = "thanos %s is initializing"
+)
+
+// Prober represents health and readriness status of given compoent.
+type Prober struct {
+	logger       log.Logger
+	loggerMtx    sync.Mutex
+	componentMtx sync.Mutex
+	component    component.Component
+	readyMtx     sync.Mutex
+	readiness    error
+	healthyMtx   sync.Mutex
+	healthiness  error
+}
+
+// SetLogger sets logger used by the Prober.
+func (p *Prober) SetLogger(logger log.Logger) {
+	p.loggerMtx.Lock()
```

Comment:
Is that really need to be thread-safe? I would remove this mutex. No one need to change this in go routines IMO.

Reply:
ok will remove